### ISSUE TITLE
[Feature Request] Make v-carousel-item more generic  

### ISSUE THREAD

**Issue by titouancreach** [vuetifyjs/vuetify#2843]
### New Functionality
`v-carousel-item` is used to display a list of image. Btw,  `src` is a required prop. I would make the component more generic using the slots. What if we want to integrate an SVG or whatever component which is not an image ?  

The carousel size if fixed to 500px, what about let the component take its children size ? In this case, the carousel would just be a container component with transition and controls. 

### Improvements
It'll let the user entirely describe what he wants in its carousel. 
```
<v-carousel>
  <v-carousel-item>
    <v-card><img /></v-card>
  </v-carousel-item>
</v-carousel>
```
I would imagine a `<v-carousel-img>` component for image intergration facilities. 

### Bugs or Edge Cases it Helps Avoid
No potential bugs just more flexibility

**Comment by jonathanbarthlm** [comment]
+1

**Comment by Carlos-Henreis** [comment]
could someone tell me how do I make a v-cards carousels?

**Comment by johnleider** [comment]
We have just done a very nice refactor on `v-tabs` and its functionality might become more generalized and used for both components as they are similar in nature. Bottom line, will be doing this.

**Comment by johnleider** [comment]
This can now be implemented with `v-window` in v1.3. Carousel will remain scoped to displaying images.
